cdc/processor/sourcemanager/engine/mounted_iter.go

```diff
@@ -47,54 +46,59 @@ func NewMountedEventIter(
 
 // Next returns the next mounted event.
 func (i *MountedEventIter) Next(ctx context.Context) (event *model.PolymorphicEvent, txnFinished Position, err error) {
+	// There are no events in mounting. Fetch more events and mounting them.
+	// The batch size is determined by `maxBatchSize`.
+	if i.nextToEmit >= len(i.rawEvents) {
+		err = i.readBatch(ctx)
+		if err != nil {
+			return
+		}
+	}
+
 	// Check whether there are events in mounting or not.
-	for idx := i.nextToEmit; idx < i.nextToMount; idx++ {
+	if i.nextToEmit < len(i.rawEvents) {
```

Comment:
It's better to use a `for` loop to call `WaitFinished` as soon as possible to release resources.